### PR TITLE
[14.0][ADD] new module:  project_forecast_line_deadline

### PR DESCRIPTION
**Pull request by ntsirintanis**

### CODE REVIEW

File: project_forecast_line_deadline/models/project_task.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    # Make forecast end date computable and stored
+    forecast_date_planned_end = fields.Date(
+        compute="_compute_forecast_date_planned_end", inverse=lambda x: None, store=True
+    )
+
+    @api.depends("date_deadline")
+    def _compute_forecast_date_planned_end(self):
+        """Set forecast end to be equal with date deadline"""
+        for this in self:
```
Comment (leemannd) [inline project_forecast_line_deadline/models/project_task.py]: It works but I am more used to `for task in self:`
I find it more readable and is the usual convention in Odoo

File: project_forecast_line_deadline/models/project_task.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    # Make forecast end date computable and stored
+    forecast_date_planned_end = fields.Date(
+        compute="_compute_forecast_date_planned_end", inverse=lambda x: None, store=True
```
Comment (leemannd) [inline project_forecast_line_deadline/models/project_task.py]: Is the inverse method mandatory?

**Review by gjotten — APPROVED** [review]
Functionally this works splendidly for us.

**Comment by leemannd** [comment]
@yankinmax do you have time to have a look at it?

**Review by yankinmax — APPROVED** [review]
LGTM, though I don't see practical purpose of this module. Just having one more same datetime?

**Comment by yankinmax** [comment]
@ntsirintanis pls squash commit into "[14.0][ADD] project_forecast_line_deadline

**Comment by gjotten** [comment]
> LGTM, though I don't see practical purpose of this module. Just having one more same datetime?

@yankinmax after porting and starting to use `project_forecast_line` for our own purposes, we found that automating the setting of both `forecast_date_planned_end` and `_start` was very useful in avoiding (too much) manual intervention.

Start date is easily set with an Automated Action of course, but for the end date we really wanted a combination of Deadline, Priority, and Milestone target dates to auto-control our planning. See https://github.com/OCA/project/pull/1248 and https://github.com/OCA/project/pull/1249 

Imagine working with internal and external projects where:
- some single Tasks have a dedicated Deadline (say, a tax filing date); we automate using Deadlines for planning
- some Projects have an agile-type setup using Milestones as containers for tasks, with a target date set - these are then picked up as end dates for planning
- finally there's (a set of) Projects where no separate Deadlines or Milestones are applicable: configurable end dates or periods for those can be set for more coarse-grained planning

**Comment by yankinmax** [comment]
@gjotten
First of all, thanks for your time to describe all these cool features and possibilities you implement on your projects.
Sounds really great.
I'll keep in mind those modules you've introduced and probably we will forward port them to recent versions.
The way `project_forecast_line` is evolving is impressive. Many thanks to your team.

**Review by leemannd — APPROVED** [review]

**Pull request merged**